app/ide-desktop/lib/client/src/security.ts

```diff
@@ -0,0 +1,114 @@
+/** @file Security configuration of Electron. Most of the options are based on the official security
+ * guide: https://www.electronjs.org/docs/latest/tutorial/security */
+
+import Electron from 'electron'
+
+// =================
+// === Constants ===
+// =================
+
+/** The list of hosts that the app can access. They are required for user authentication to work. */
+const trustedHosts = ['accounts.google.com', 'accounts.youtube.com', 'github.com']
+
+/** The list of URLs a new WebView can be pointed to. */
+const webViewUrlWhitelist: string[] = []
+
+// =============
+// === Utils ===
+// =============
+
+export function secureWebPreferences(webPreferences?: Electron.WebPreferences) {
+    if (!webPreferences) {
+        webPreferences = {}
+    }
+    delete webPreferences.preload
+    delete webPreferences.nodeIntegration
+    delete webPreferences.nodeIntegrationInWorker
+    delete webPreferences.webSecurity
+    delete webPreferences.allowRunningInsecureContent
+    delete webPreferences.experimentalFeatures
+    delete webPreferences.enableBlinkFeatures
+    delete webPreferences.contextIsolation
```

Comment:
TBH it's not clear to me why deleting settings make it secure. Shouldn't we override them to "secure" values?